### PR TITLE
feat: allow docx importer implementation via external service

### PR DESCRIPTION
**Pull request by mussbach**
This PR adds an alternative way to import docx via an external api service. The existing implementation via rabbitmq is still the default one, it is possible to switch via Interface implementation.

### How to review/test
code review might be best. If you have access to the new service you could use it. 

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterHttp.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+            $result = [
+                'procedure'  => $procedure,
+                'category'   => $category,
+                'elementId'  => $elementId,
+                'path'       => $file->getRealPath(),
+                'paragraphs' => Json::decodeToArray($response->getContent()),
+            ];
+        } catch (Exception $e) {
+            $this->logger->error('Error while creating pdf with http: '.$e->getMessage());
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Tools/DocxImporterHttp.php:55]: Is pdf correct here? I don't see the creation of an pdf.

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+     * @var RpcClient
+     */
+    protected $client;
+
+    public function __construct(private readonly LoggerInterface $logger, private readonly GlobalConfigInterface $globalConfig)
+    {
+    }
+
+    /**
+     * Importiere ein docx-Dokument mittels RabbitMQ Instanz.
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php:36]: A german comment

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+     *
+     * @throws Exception
+     */
+    public function importDocx(File $file, string $elementId, string $procedure, string $category): array
+    {
+        try {
+            // Generiere Message
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php:43]: german

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                'elementId' => $elementId,
+                'path'      => $file->getRealPath(),
+            ]);
+
+            $routingKey = $this->globalConfig->getProjectPrefix();
+            if ($this->globalConfig->isMessageQueueRoutingDisabled()) {
+                $routingKey = '';
+            }
+
+            // Füge Message zum Request hinzu
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php:56]: german

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                'Import docx with RabbitMQ, with routingKey: '.$routingKey
+            );
+            $this->client->addRequest(
+                $msg,
+                'importDemosPlan',
+                'import',
+                $routingKey,
+                300
+            );
+            // Anfrage absenden
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php:67]: german

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            if ('' != $replies['import']) {
+                $this->logger->info(
+                    'Incoming message size:'.strlen($replies['import'])
+                );
+            }
+
+            return Json::decodeToArray($replies['import']);
+        } catch (AMQPTimeoutException $e) {
+            $this->logger->error(
+                'Fehler in ImportConsumer:',
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php:79]: german words in error message

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+                [$e]
+            );
+            throw new TimeoutException($e->getMessage());
+        } catch (Exception $e) {
+            $this->logger->error(
+                'Fehler in ImportConsumer:',
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php:85]: german

**Review by breierDemos — CHANGES_REQUESTED** [review]
One question and some german words that should be translated to english I think.

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterHttp.php (hunk 1 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Tools/DocxImporterHttp.php]: I just moved the code, but it is better to improve the wordings, thx adb552ab

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php (hunk 2 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php]: adb552ab

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php (hunk 3 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php]: adb552ab

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php (hunk 4 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php]: adb552ab

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php (hunk 5 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php]: adb552ab

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php (hunk 6 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php]: adb552ab

File: demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php (hunk 7 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Tools/DocxImporterRabbitmq.php]: adb552ab

**Review by breierDemos — APPROVED** [review]
lgtm code review only

**Review by graupnerdemos — DISMISSED** [review]
I for one can't wait to refactor this to symfony/messenger but for the time being, this looks good.

**Comment by mussbach** [comment]
fyi: I changed the base as the problem I needed this in the releasebranch could be solved otherwise, so we have more time to test this one

**Pull request merged**